File: src/main/java/com/musinsa/subject/service/facade/ProductFacade.java
package com.musinsa.subject.service.facade;

import com.musinsa.subject.model.product.ProductRequest;
import com.musinsa.subject.model.product.ProductResponse;

public interface ProductFacade {

    ProductResponse createProduct(ProductRequest request);

    ProductResponse getProduct(long productId);

    ProductResponse updateProduct(long productId, ProductRequest request);

    void deleteProduct(long productId);
    
}

File: src/main/kotlin/com/musinsa/subject/service/facade/ProductFacade.kt
package com.musinsa.subject.service.facade

import com.musinsa.subject.model.product.ProductRequest
import com.musinsa.subject.model.product.ProductResponse

interface ProductFacade {
    fun createProduct(request: ProductRequest): ProductResponse
    fun getProduct(productId: Long): ProductResponse
    fun updateProduct(productId: Long, request: ProductRequest): ProductResponse
    fun deleteProduct(productId: Long)
}
